### DISCUSSION TITLE
How do i use php mailer with cpanel

### DISCUSSION THREAD

**Discussion by CaptainEboy** [PHPMailer/PHPMailer#2331] [Q&A] [answered]
I can't seem to find any documentation on how to use php mailer on an hosted code using cpanel.

References, links & explanations along with working & testable code snippets are greatly appreciated

**Comment by Synchro** [comment, marked as answer]
I've no idea about cpanel specifically, but you'd run it like you run any other PHP code: Configure a web server with a PHP runtime (usually PHP-FPM), point it at your scripts and off you go. How you do that it outside PHPMailer's scope. There is a ton of example code provided in the examples folder; I don't think you would need to do anything special at all to make all of them work with cpanel.

**Reply by CaptainEboy** [reply]
I think the main problem I have is with installing composer, don't really understand it.

I am actually new to php, more of a js developer

**Reply by Synchro** [reply]
composer is to PHP what npm is to js; a package manager. It's a command-line tool to download, install, and manage your dependencies, as PHPMailer might be for your project. It's generally easier and more reliable to use composer than to load a library yourself, for even a single dependency. A typical setup (after you have installed PHP and composer on your development machine) is simply:

    composer require phpmailer/phpmailer;

This will create a `composer.json` file that lists your dependencies (the equivalent of npm's `packages.json`), a `composer.lock` file that records *exactly* which versions of each package you're using so that you can use it to install the same packages elsewhere (e.g. on your servers), and a `vendor` folder which contains composer's own libraries, your packages (such as PHPMailer), and a file called `autoload.php` that deals with finding and loading your dependencies whenever you want to use them. To use all of that, all you need in your application is:

    require 'vendor/autoload.php';

and it will automatically deal with loading packages for you, so you can just say `$mail = new PHPMailer();` without having to `require` it manually beforehand. You will still need to write your own `use` statements though.

Having done all that, when it comes to deployment, you can either run `composer install` on your server to install the same dependencies, or if you can't run CLI code on your server (e.g. you're on shared hosting), upload the `vendor` folder along with all your other PHP files.

Beyond this, this isn't reallly the place to answer how to use composer or how to write PHP.

**Reply by CaptainEboy** [reply]
Thanks alot.

Just sent my first email with phpmailer✌😁💥💖.

I didn't actually use composer.

Seems Cpanel blocks/doesn't work with alot of code on the main documentation.

I think the documentations should be updated

**Reply by Synchro** [reply]
I agree, cpanel should document how they prevent you from running your code...

**Reply by gaffordRich** [reply]
CaptainEboy where you able to make use of Cpanel in sending email with PHPmailer? im currently having difficulty with this myself though i intend to use mine to send SMS text, kindly help out
